src/ui/handler/mouse.js

```diff
@@ -6,6 +6,17 @@ import type Point from '@mapbox/point-geometry';
 const LEFT_BUTTON = 0;
 const RIGHT_BUTTON = 2;
 
+// the values for each button in MouseEvent.buttons
+const BUTTONS_FLAGS = {
+    [LEFT_BUTTON]: 1,
+    [RIGHT_BUTTON]: 2
+};
+
+function buttonStillPressed(e: MouseEvent, button: number) {
+    const flag = BUTTONS_FLAGS[button];
+    return e.buttons === undefined || (e.buttons & flag) !== flag;
```

Comment:
Is it possible to press the left button, move the mouse outside the window, release the button, then click the left button again and bring it back inside the window so that this isn't triggered? Or does releasing the button always make `e.buttons` undefined?

Reply:
> Is it possible to press the left button, move the mouse outside the window, release the button, then click the left button again and bring it back inside the window so that this isn't triggered?
I think it might be possible but I wasn't able to trigger it. With both ctrl-click and right click that starts outside the browser it only starts firing mousemoves back in the window after mouseup. But I think if it did happen it should be ok. The gesture would continue where the user left off.

> Or does releasing the button always make e.buttons undefined?

`e.buttons` got added in safari in 11. To handle those browsers more gracefully it's saying "yes the button is pressed" in those cases even if we can't tell. If this bug were present in those browsers it would remain but everything else would still work.

Reply:
And as you noted it's not present in android chrome at all. This would handle that gracefully if you had a mouse connected to Android.